alphadia/workflow/peptidecentric/optimization_handler.py

```diff
@@ -0,0 +1,445 @@
+import numpy as np
+import pandas as pd
+from alphabase.spectral_library.base import SpecLibBase
+
+from alphadia.constants.settings import MAX_FRAGMENT_MZ_TOLERANCE
+from alphadia.data.alpharaw_wrapper import AlphaRaw
+from alphadia.data.bruker import TimsTOFTranspose
+from alphadia.reporting.reporting import Pipeline
+from alphadia.workflow import optimization
+from alphadia.workflow.config import Config
+from alphadia.workflow.managers.calibration_manager import CalibrationManager
+from alphadia.workflow.managers.fdr_manager import FDRManager
+from alphadia.workflow.managers.optimization_manager import OptimizationManager
+from alphadia.workflow.peptidecentric.extraction_handler import ExtractionHandler
+from alphadia.workflow.peptidecentric.recalibration_handler import RecalibrationHandler
+from alphadia.workflow.peptidecentric.utils import fdr_correction, log_precursor_df
+
+
+class OptimizationHandler:
+    """
+    Handles the optimization of peptide-centric workflows.
+    """
+
+    def __init__(
+        self,
+        config: Config,
+        optimization_manager: OptimizationManager,
+        calibration_manager: CalibrationManager,
+        fdr_manager: FDRManager,
+        extraction_handler: ExtractionHandler,
+        recalibration_handler: RecalibrationHandler,
+        reporter: Pipeline,
+        spectral_library: SpecLibBase,
+        dia_data: AlphaRaw | TimsTOFTranspose,
+    ):
+        self.config = config
+        # TODO decouple optimizers from workflow:
+        #  there's an implicit coupling between this class and the Optimizers, which think this is a workflow object!
+        self.optimization_manager = optimization_manager
```

Comment:
I think here we really define the code pattern better. I would be very confused if there are both optimization handler and optimization manager :D

Reply:
I think making this an `OptimizationContext` would be the cleanest